### ISSUE TITLE
debian/stretch APT repo is no longer online

### ISSUE THREAD

**Issue by stromfinger** [g7/sailfishx-patcher-f5321#31]
If I try to patch the current Sailfish-OS version V 4.5.0.21 (Jul. 2023), then the Vagrant VM up-start writes an error message and breaks:
```
    sailfish_image_patcher: E: The repository 'http://security.debian.org/debian-security stretch/updates Release' does no longer have a Release file.
    sailfish_image_patcher: E: The repository 'http://deb.debian.org/debian stretch Release' does no longer have a Release file.
```

When I update the vagrantfile line 46 to "config.vm.box = "debian/buster64", the vagrant environment bring up successful but write out few minor error messages (lines 450f, 529, 695ff, 702ff, 712ff and 719ff see logfile attached).

[vagrant_up_with_debian-buster64.txt](https://github.com/g7/sailfishx-patcher-f5321/files/12059071/vagrant_up_with_debian-buster64.txt)

The final patched image of SFOS V4.5.0.21 is running successful at F5321.

The patch is running at Linux Mint Debian Edition 5.

**Comment by g7** [comment]
Thanks for the bug report, unfortunately Debian moved the stretch repositories to archive.debian.org since it's not supported anymore.

I'll happily merge a pull request that bumps the debian version to buster or perhaps even better bullseye or bookworm. Unfortunately I would not be able to test that on device (but I'm glad that after all those years the patch and the compatibility layer still works!).

**Comment by henning-schild** [comment]
i might be able to help on that but can not promise if or when, i still have a spare device that i could use for such stuff

**Comment by NicePics** [comment]
Vagrant is not even necessary as you can run the patcher directly from your current system as long as you have the right packages installed (simg2img, img2simg, rsync, pigz, losetup, pvcreate, vgcreate, e2fsck, resize2fs).
On Slackware I just had to install the android tools and pigz and then comment out the following part from patch.sh:
`#if [ "$ARCH" != "armv7l" ] || [ "$ARCH" != "aarch64" ]; then
#	WITH_QEMU_STATIC="yes"
#	check_application qemu-arm-static
#fi`

**Comment by NicePics** [comment]
Vagrant is not really necessary - just use your current system, install the deps listed in patch.sh (android tools, qemu, lvm etc) and run the script directly! Just got Sailfish OS 4.5 Struven Ketju made this way.

**Comment by stromfinger** [comment]
I can help testing the patching process and its artifacts too. I have a second device F5321 using for vatious tests with a SIM slot failure.

**Comment by g7** [comment]
Thank you folks for the feedback, I have pushed [b3b7d98](https://github.com/g7/sailfishx-patcher-f5321/commit/b3b7d980da02f69df13d442afcf1140a288588e2) in branch `version-bump`. I would appreciate if you could test it when you have time :)

Yes, vagrant is not necessary, but useful to have something reproducible also on other platforms as well. Glad to hear that 4.5 still patches fine ;)

**Comment by stromfinger** [comment]
I've tested the branch `version-bump` with SFOS image 4.5.0.21 now, but it is running into an error just at the begin of patch process (`vagrant ssh...`):
```
W: You must be root to use this script! Trying with sudo...
E: qemu-arm-static is missing. Please install the relative package and retry
```

When I start the patch process as root (sudo), the message is a bit more verbose:
```
Vagrant failed to initialize at a very early stage:

The plugins failed to initialize correctly. This may be due to manual
modifications made within the Vagrant home directory. Vagrant can
attempt to automatically correct this issue by running:

  vagrant plugin repair

If Vagrant was recently updated, this error may be due to incompatible
versions of dependencies. To fix this problem please remove and re-install
all plugins. Vagrant can attempt to do this automatically by running:

  vagrant plugin expunge --reinstall

Or you may want to try updating the installed plugins to their latest
versions:

  vagrant plugin update

Error message given during initialization: Unable to resolve dependency: user requested 'vagrant-vbguest (= 0.16.0)'
```
I'm sure, that I have installed 'vbguest' before by `vagrant plugin install vagrant-vbguest` but this command has installed version 0.31.0:
```
Installing the 'vagrant-vbguest' plugin. This can take a few minutes...
Installed the plugin 'vagrant-vbguest (0.31.0)'!
```
Any command `vagrant plugin...` doesn't fix the error.

I have logged the output of command `vagrant up` before to enable you to debug this issue - see file [vagrant_up_bookworm64.log](https://github.com/g7/sailfishx-patcher-f5321/files/12234683/vagrant_up_bookworm64.log).

On the other hand, I have tested the whole patching process by using `buster64` and `bullseye64` includes testing the image artifact by flashing to a real F5321 successful. There are the log files for `vagrant up` and `vagrant ssh...` commands:

- [vagrant_up_buster64.log](https://github.com/g7/sailfishx-patcher-f5321/files/12234720/vagrant_up_buster64.log)
- [vagrant_ssh_buster64.log](https://github.com/g7/sailfishx-patcher-f5321/files/12234726/vagrant_ssh_buster64.log)
- [vagrant_up_bullseye64.log](https://github.com/g7/sailfishx-patcher-f5321/files/12234728/vagrant_up_bullseye64.log)
- [vagrant_ssh_bullseye64.log](https://github.com/g7/sailfishx-patcher-f5321/files/12234731/vagrant_ssh_bullseye64.log)

Finally a line about my test system (uname -a)
```
Linux tokje 5.10.0-23-amd64 #1 SMP Debian 5.10.179-1 (2023-05-12) x86_64 GNU/Linux
```

**Comment by g7** [comment]
@stromfinger thank you very much for the thorough feedback, much appreciated! I pushed a change that should make `vagrant up` work correctly now, please test when you have time :)

Thanks!

**Comment by stromfinger** [comment]
@g7 just as the patching process with the last commit 6d6f6a6 has run successfully as the patched artifact (SFOS 4.5.0.21 on F5321) is running with no problems - thank you for the good work. I put the log files here, so you has a chance to reflect the result:

- [vagrant_up_6d6f6a6.log](https://github.com/g7/sailfishx-patcher-f5321/files/12306902/vagrant_up_6d6f6a6.log)
- [vagrant_ssh_6d6f6a6.log](https://github.com/g7/sailfishx-patcher-f5321/files/12306905/vagrant_ssh_6d6f6a6.log)

Thank you!

Edit: typo

**Comment by g7** [comment]
@stromfinger thank you very much for testing! :) Merging!
